tasks/elasticsearch-plugins.yml

```diff
@@ -46,18 +46,20 @@
   environment:
     CONF_DIR: "{{ conf_dir }}"
     ES_INCLUDE: "{{ instance_default_file }}"
+    ES_ENV_FILE: "{{ instance_default_file }}"
 
 - name: Install elasticsearch plugins
-  command: "{{es_home}}/bin/elasticsearch-plugin install {{ item.plugin }} --batch --silent"
+  command: "{{es_home}}/bin/elasticsearch-plugin install {{ item.url | default(item.plugin) }} --batch --silent"
   register: plugin_installed
   failed_when: "'ERROR' in plugin_installed.stdout"
   changed_when: plugin_installed.rc == 0
   with_items: "{{ es_plugins }}"
-  when: item.plugin in plugins_to_install
+  when: item.plugin is defined and item.plugin in plugins_to_install
```

Comment:
@orphaner I think it's better to add default value to avoid looping if empty.

```yml
  with_items: "{{ es_plugins | default([]) }}"
  when: item.plugin in plugins_to_install
```

Reply:
@barryib i think we need the additional check incase people forget to specify the plugin var so...

```
with_items: "{{ es_plugins | default([]) }}"
when: item.plugin is defined and item.plugin in plugins_to_install
```
?

Reply:
@gingerwizard Right. So, LGTM